src/web/releases.rs

```diff
@@ -770,15 +756,145 @@ mod tests {
+        wrapper(|env| {
+            let db = env.db();
+            db.fake_release().name("regex").version("0.0.0").create()?;
+
+            let (num_results, results) = get_search_results(&db.conn(), "redex", 1, 100);
+            assert_eq!(num_results, 1);
+
+            let mut results = results.into_iter();
+            assert_eq!(results.next().unwrap().name, "regex");
+            assert_eq!(results.count(), 0);
+
+            Ok(())
+        })
+    }
+
+    #[test]
+    fn search_descriptions() {
+        wrapper(|env| {
+            let db = env.db();
+            db.fake_release()
+                .name("something_completely_unrelated")
+                .description("Supercalifragilisticexpialidocious")
+                .create()?;
+
+            let (num_results, results) =
+                get_search_results(&db.conn(), "supercalifragilisticexpialidocious", 1, 100);
+            assert_eq!(num_results, 1);
+
+            let mut results = results.into_iter();
+            assert_eq!(
+                results.next().unwrap().name,
+                "something_completely_unrelated"
+            );
+            assert_eq!(results.count(), 0);
+
+            Ok(())
+        })
+    }
+
+    #[test]
+    fn search_limits() {
+        wrapper(|env| {
+            let db = env.db();
+
+            db.fake_release().name("something_magical").create()?;
+            db.fake_release().name("something_sinister").create()?;
+            db.fake_release().name("something_fantastical").create()?;
+            db.fake_release()
+                .name("something_completely_unrelated")
+                .create()?;
+
+            let (num_results, results) = get_search_results(&db.conn(), "something", 1, 2);
+            assert_eq!(num_results, 4);
+
+            let mut results = results.into_iter();
+            assert_eq!(results.next().unwrap().name, "something_magical");
+            assert_eq!(results.next().unwrap().name, "something_sinister");
+            assert_eq!(results.count(), 0);
+
+            Ok(())
+        })
+    }
+
+    #[test]
+    fn search_offsets() {
+        wrapper(|env| {
+            let db = env.db();
+            db.fake_release().name("something_magical").create()?;
+            db.fake_release().name("something_sinister").create()?;
+            db.fake_release().name("something_fantastical").create()?;
+            db.fake_release()
+                .name("something_completely_unrelated")
+                .create()?;
+
+            let (num_results, results) = get_search_results(&db.conn(), "something", 2, 2);
+            assert_eq!(num_results, 4);
+
+            let mut results = results.into_iter();
+            assert_eq!(results.next().unwrap().name, "something_fantastical");
+            assert_eq!(
+                results.next().unwrap().name,
+                "something_completely_unrelated"
+            );
+            assert_eq!(results.count(), 0);
+
+            Ok(())
+        })
+    }
```

Comment:
Can you add tests for
- ordering by release date
- fuzzy name search taking precedence over description search
- unrelated crates not being returned
- github stars (optional)

Reply:
Didn't do github stars, as I don't sort by them